tests/unit/test_contracts/kv_make_key_tests.cpp

```diff
@@ -0,0 +1,296 @@
+};
+
+struct my_table : eosio::kv_table<my_table, my_struct, "testtable"_n, "eosio.kvram"_n> {
+   struct {
+      kv_index tname{&my_struct::tname};
+      kv_index tstring{&my_struct::tstring};
+      kv_index tui64{&my_struct::tui64};
+      kv_index ti32{&my_struct::ti32};
+      kv_index ti128{&my_struct::ti128};
+      kv_index tfloat{&my_struct::tfloat};
+      kv_index tstruct{&my_struct::tstruct};
+      kv_index ttuple{&my_struct::ttuple};
+      kv_index itstring{&my_struct::itstring};
+   } index;
+
+   my_table(eosio::name contract_name) {
+      init(contract_name, &index);
+   }
+};
+
+class [[eosio::contract]] kv_make_key_tests : public eosio::contract {
+public:
+   using contract::contract;
+   my_struct s1{
+      .tname = "bob"_n,
+      .tstring = "a",
+      .tui64 = 5,
+      .ti32 = 0,
+      .ti128 = (static_cast<uint128_t>(1) << 127) - 5,
+      .tfloat = 4.2574,
+      .tstruct = { 1, 2 },
+      .ttuple = { 100, 32.43, "abc"}
+   };
+   my_struct s2{
+      .tname = "alice"_n,
+      .tstring = "C",
+      .tui64 = 4,
+      .ti32 = -1,
+      .ti128 = (static_cast<uint128_t>(1) << 127) - 4,
+      .tfloat = 5.2574,
+      .tstruct = { 5, 6 },
+      .ttuple = { 100, 32.43, "def"}
+   };
+   my_struct s3{
+      .tname = "john"_n,
+      .tstring = "e",
+      .tui64 = 3,
+      .ti32 = -2,
+      .ti128 = (static_cast<uint128_t>(1) << 127) - 3,
+      .tfloat = 187234,
+      .tstruct = { 3, 4 },
+      .ttuple = { 100, 33.43, "abc"}
+   };
+   my_struct s4{
+      .tname = "joe"_n,
+      .tstring = "g",
+      .tui64 = 2,
+      .ti32 = 1,
+      .ti128 = (static_cast<uint128_t>(1) << 127) - 2,
+      .tfloat = 0,
+      .tstruct = { 7, 8 },
+      .ttuple = { 101, 32.43, "abc"}
+   };
+   my_struct s5{
+      .tname = "billy"_n,
+      .tstring = "I",
+      .tui64 = 1,
+      .ti32 = 2,
+      .ti128 = (static_cast<uint128_t>(1) << 127) - 1,
+      .tfloat = -4.2574,
+      .tstruct = { 9, 10 },
+      .ttuple = { 101, 34.43, "abc"}
+   };
+
+   [[eosio::action]]
+   void setup() {
+      my_table t = my_table::open("kvtest"_n);
+
+      t.put(s1);
+      t.put(s2);
+      t.put(s3);
+      t.put(s4);
+      t.put(s5);
+   }
+
+   [[eosio::action]]
+   void makekeyname() {
+      my_table t = my_table::open("kvtest"_n);
+
+      auto end_itr = t.index.tname.end();
+      auto itr = t.index.tname.begin();
+
+      eosio::check(itr != end_itr, "Should not be the end");
+      eosio::check(itr.value().tname == s2.tname, "Got the wrong value");
+      ++itr;
+      eosio::check(itr.value().tname == s5.tname, "Got the wrong value");
+      ++itr;
+      eosio::check(itr.value().tname == s1.tname, "Got the wrong value");
+      ++itr;
+      eosio::check(itr.value().tname == s4.tname, "Got the wrong value");
+      ++itr;
+      eosio::check(itr.value().tname == s3.tname, "Got the wrong value");
+      ++itr;
+      eosio::check(itr == end_itr, "Should be the end");
```

Comment:
This is very repetitive.  If you compare the entire struct instead of just the specific key, you can define a helper function that can be called like:
```
check_index(t.index.tname, {s2, s5, s1, s4, s3});
```

Reply:
This cleaned it up a lot, thanks.